sklearn/preprocessing/_csr_polynomial_expansion.pyx

```diff
@@ -43,11 +52,14 @@ cdef inline INDEX_T _deg3_column(INDEX_T d, INDEX_T i, INDEX_T j, INDEX_T k,
                 + d * j + k)
 
 
-def _csr_polynomial_expansion(cnp.ndarray[DATA_T, ndim=1] data,
-                              cnp.ndarray[INDEX_T, ndim=1] indices,
-                              cnp.ndarray[INDEX_T, ndim=1] indptr,
-                              INDEX_T d, INDEX_T interaction_only,
-                              INDEX_T degree):
+def _csr_polynomial_expansion(
+    DATA_T[:] data,
```

Comment:
To support memory mapped or read-only array, we need to use `const`-qualified memoryviews. Unfortunately this can't be made possible when using fuse-types for Cython<3.0.

Moreover, https://github.com/scikit-learn/scikit-learn/pull/23731/ is going to soon perform notable changes to this file and your work will need to be updated then.

Hence, I think this PR is unfortunately blocked for now.

Reply:
Right data is a read only array. Should I close this PR for now and reopen it when applicable?